File: js Practice/anagram.js
anagram=(word1,word2)=>{

    let wordarray1=word1.split('');
    let wordarray2=word2.split('');

    let frequencycounyter1={}
    let frequencycounyter2={}

    if(wordarray1.length!=wordarray2.length){
        return false;
    }

    for(let val of wordarray1){
        frequencycounyter1[val]=(frequencycounyter1[val]||0)+1;
    }

    for(let val of wordarray2){
        frequencycounyter2[val]=(frequencycounyter2[val]||0)+1;
    }

    console.log(wordarray1)
    console.log(wordarray2)
    console.log(frequencycounyter1)
    console.log(frequencycounyter2)

    for(let key in frequencycounyter1){

                if(frequencycounyter1[key]!==frequencycounyter2[key]){
            return false;
        }
    }

    return true;

}

File: java Practice/Anagram.java
import java.io.*;
import java.util.Arrays;

public class Anagram {

    public static void main(String[] args) throws IOException {

        BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));
        String[] word1 = bf.readLine().split("");
        String[] word2 = bf.readLine().split("");

        Arrays.sort(word1);
        Arrays.sort(word2);

        String reword1 = "";
        String reword2 = "";

        if (word1.length != word2.length) {
            System.out.println("Not Anagram");
        } else {

            for (int i = 0; i < word1.length; i++) {

                reword1 += word1[i];

            }

            for (int j = 0; j < word2.length; j++) {

                reword2 += word2[j];
            }

            if (reword1.equalsIgnoreCase(reword2)) {
                System.out.println("Anagram");
            } else {

                System.out.println("not Anagram");

            }

        }
    }

}
